### DISCUSSION TITLE
Reset Vector Storage not working properly

### DISCUSSION THREAD

**Discussion by bgeneto** [open-webui/open-webui#5558] [General]
# Bug Report

## Installation Method

Docker with docker-compose

## Environment

- **Open WebUI Version:** v0.3.22
- **Ollama (if applicable):** n/a

- **Operating System:** Debian 12 kernel 6.1.106-3
- **Browser (if applicable):** n/a

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

When we go to Admin -> Settings -> Documents when click "Reset Vector Storage" I expect that it empties the `chroma.sqlite3` file and delete all the documents folder in `vector_db` directory. 
Also, when we delete a document via Workspace -> Documents, the corresponding entries in the vector storage should be deleted. 

## Actual Behavior:

When we go to Admin -> Settings -> Documents when click "Reset Vector Storage" it does touches the `chroma.sqlite3` file but its size does not decreases and do not delete the corresponding documents folder in `vector_db` directory. It seems to me that this button is not working properly. 
Also, when we delete a document via Workspace -> Documents, the corresponding entries in the vector storage seem not to be deleted. 

## Description

**Bug Summary:**

Reset Vector Storage not working properly.

## Reproduction Details

**Steps to Reproduce:**
Just click Reset Vector Storage and check `chroma.sqlite3` file and `vector_db` directory.

## Logs and Screenshots

**Browser Console Logs:**
n/a

**Docker Container Logs:**
n/a

**Screenshots/Screen Recordings (if applicable):**
n/a

**Comment by mesa57** [comment]
I can confirm this behaviour. There should be a way to cleanup obsolete documents and their data in the vector database.

**Comment by zouzeTG** [comment]
You have to shrink the database, is possible to automatized  : You can read this :  https://stackoverflow.com/questions/21668969/whats-the-effective-way-to-compact-sqlite3-db

**Reply by bgeneto** [reply]
Yeah... Thx for reference. The idea here is that open-webui should set to auto compact database, or use VACUUM every time we click reset dB, and not leave that responsibility to the user with command line access to the server.

**Reply by spammenotinoz** [reply]
> You have to shrink the database, is possible to automatized : You can read this : https://stackoverflow.com/questions/21668969/whats-the-effective-way-to-compact-sqlite3-db

No that is not the issue, "the reset" feature doesn't delete the objects from either the database or storage so a Vaccuum is pointless.

**Comment by mesa57** [comment]
This issue is not resolved by that action

> Also, when we delete a document via Workspace -> Documents, the corresponding entries in the vector storage seem not to be deleted

**Comment by spammenotinoz** [comment]
As the reset is not deleting the files from disk or the database, database maintenance like a vaccuum does not address the problem.
Does anyone have a workaround?  The below worked for me but it is brutal.
- Stop the App
- Delete all folders in the vector_db folder, including and the vector DB
- Start Open-WebUI

**Reply by Classic298** [reply]
Yeah that works but it is indeed brutal and you need to do it manually every now and then.

Why is openwebui not capable of identifying the very documents I uploaded and then deleting them when I also delete the chat? If I delete the chat -> then also it should delete all the files i uploaded from the upload folder and from the vector db.

**Reply by McNickSistoPro** [reply]
@Classic298 indeed, I 100% second this.

**Comment by gabriel-ecegi** [comment]
Hi, same problem here. 
I have noticed, it works properly when server isn't restarted. 

But the files in vector_db are not deleted when i upload files to knowledge, restart the python server, click on "Reset Upload Directory" and "Reset Vector Storage/Knowledge".

**Comment by Classic298** [comment]
Hi, same Problem.

1) reset vector storage doesn't do anything 
2) files are never deleted from vector storage even if the chat they were uploaded into doesn't even exist anymore

Why is this not a reported issue and instead here in the discussion??

**Comment by eureka-0** [comment]
Hi, same here.

I uploaded three documents (~5MB) and the vector_db directory grew signicicantly (~400MB). Is there a safe way to clear the vector_db, given that this option is broken?

**Comment by McNickSistoPro** [comment]
Hi, same problem here using v0.5.16 but it seems to happen also on the v0.5.19.

**Comment by Constey** [comment]
Aswell in 0.5.20 and 0.6.

**Comment by jetnet** [comment]
Seems to be not that important one :(

**Comment by ZombieLurker** [comment]
One of the main purposes of using OWUI is for Local LLM's, and one of the main reasons for using Local LLM's is for data privacy. The fact things are not being deleted and devs are not responding to this discussion after this long? Is a bit insane and I guess now after finding this out, I won't be using OWUI anymore if they don't implement a way to actually delete conversation and upload data swiftly. This discussion has been open for too long without a response. 

Seems like they are just trying ot hide this from users cause I assume most people think that when they deleted the chat history, the chat and everything included in it is actually DELETED.
